dev/environment-dev.yml

```diff
@@ -41,7 +41,8 @@ dependencies:
   - cryptography
   - securesystemslib
   # libmambapy build dependencies
-  - scikit-build
+  - scikit-build-core
+  - setuptools-scm
   - pybind11-stubgen <1.0
```

Comment:
Does conda still not have any integration with python build systems? Manually referencing them like this is very error-prone, particularly since some build dependencies are dynamically generated, `setuptools-scm` in this case.